libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1797,14 +1823,26 @@ void GCS_MAVLINK::handle_timesync(mavlink_message_t *msg)
     mavlink_timesync_t tsync;
     mavlink_msg_timesync_decode(msg, &tsync);
 
-    // ignore messages in which tc1 field (timestamp 1) has already been filled in
     if (tsync.tc1 != 0) {
-        return;
+        // this is a response to a timesync request
+        if (tsync.ts1 != _timesync_request.sent_ts1) {
+            // we didn't actually send the request.... or it's a
+            // response to an ancient request...
+            return;
+        }
```

Comment:
We should always return, according to other implementations we shouldn't answer anything when tc1 is filled. I'm wondering though if, since we are adding sending TIMESYNC messages, we should log the response.

Reply:
Fixed.  I've also added logging.